client/signup/steps/domains/index.jsx

```diff
@@ -317,30 +317,19 @@ class DomainsStep extends React.Component {
 	};
 
 	shouldIncludeDotBlogSubdomain() {
-		const { flowName, isDomainOnly, siteGoals, signupDependencies } = this.props;
-		const siteGoalsArray = siteGoals ? siteGoals.split( ',' ) : [];
-
-		// 'subdomain' flow coming from .blog landing pages
-		if ( flowName === 'subdomain' ) {
-			return true;
-		}
-
-		// 'blog' flow, starting with blog themes
-		if ( flowName === 'blog' ) {
-			return true;
-		}
-
-		// No .blog subdomains for domain only sites
-		if ( isDomainOnly ) {
+		// Disable for domain only sites
+		if ( this.props.isDomainOnly ) {
 			return false;
 		}
 
-		// If we detect a 'blog' site type from Signup data
+		// Enable for 'subdomain' flow, originating from .blog landing pages
```

Comment:
The `subdomain` flow was removed in https://github.com/Automattic/wp-calypso/pull/32505

Since `subdomain` isn't defined as a flow in [flows-pure.js](https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/flows-pure.js), anything pointing to `/start/subdomain` will revert to the default onboarding flow at `/start`.